### ISSUE TITLE
ForEachItem fails with [Cannot invoke "java.util.Map.get(Object)" because "taskOutput" is null] when there are no items to process

### ISSUE THREAD

**Issue by aku** [kestra-io/kestra#4278]
TLDR:

Here is a simple workflow you can use to reproduce the problem:

```
id: debug_each_issue
namespace: kestra_issues

tasks:

  - id: workingDir
    type: io.kestra.plugin.core.flow.WorkingDirectory
    tasks:
    - id: bash
      type: io.kestra.plugin.scripts.shell.Commands
      commands:
        - touch test
    - id: outputFiles
      type: io.kestra.plugin.core.storage.LocalFiles
      outputs:
        - test

  - id: for_each_will_fail
    type: io.kestra.plugin.core.flow.ForEachItem
    items: "{{ outputs.outputFiles.uris['test'] }}"
    batch:
      partitions: 5 
    flowId: debug_each_subflow
    namespace: kestra_issues
    wait: true
    transmitFailed: true
    inputs:
      test: "{{ taskrun.items }}"
```

```
id: debug_each_subflow
namespace: kestra_issues

inputs:
  - id: test
    type: STRING

tasks:

  - id: print_status
    type: io.kestra.plugin.core.log.Log
    message: got {{ inputs.test}}
```

<img width="1358" alt="image" src="https://github.com/kestra-io/kestra/assets/678596/ffc9acc4-4cde-4b80-8e4d-a212a47bc972">

---

### Describe the issue

UPD: it looks like the issue happens when there is no items.
I used FilteTransform that filters out some records. Probably ForEachItem did not have data to process
Still this is not expected behaviour. If there are no items to process FEI should complete without errors

---

When Kafka trigger returns only 1 record the following flow fails with 

```java.util.Map.get(Object)" because "taskOutput" is null```

Also, notice that it won't create a subflow execution.

When Kafka trigger returns multiple records, everything works.

flow:

```
....

concurrency:
  limit: 1 

tasks:
  - id: select_suppliers
    type: io.kestra.plugin.scripts.jython.FileTransform    
    from: "{{ trigger.uri }}"
    concurrent: 3
    script: |
         [rest is omitted for brevity]

  - id: verify_suppliers_in_batches
    type: io.kestra.plugin.core.flow.ForEachItem
    items: "{{ outputs.select_suppliers.uri }}"
    batch:
      partitions: 5 
    flowId: [SUBFLOW ID]
    namespace: [REDACTED]
    wait: true
    transmitFailed: true
    inputs:
      suppliers: "{{ taskrun.items }}"

  - id: purge_execution
    type: io.kestra.plugin.core.storage.PurgeExecution

pluginDefaults:
  - type: io.kestra.plugin.kafka.Trigger
    values:
      logLevel: WARN

triggers:
  - id: read_suppliers_registrations
    [rest is omitted for brevity]    
```

Where there is 1 record:

<img width="867" alt="image" src="https://github.com/kestra-io/kestra/assets/678596/afb409db-b210-4e16-a789-2e0e3bebd89d">

<img width="721" alt="image" src="https://github.com/kestra-io/kestra/assets/678596/264527c5-abe6-4d65-8c72-3c04dca31e72">

When there are > 1 records

<img width="876" alt="image" src="https://github.com/kestra-io/kestra/assets/678596/4d19104b-4638-4b88-b27d-70115b1c4611">

<img width="1347" alt="image" src="https://github.com/kestra-io/kestra/assets/678596/64827405-e0df-41a8-903e-e075b52fb3cd">

### Environment

- Kestra Version: 17.8
- Operating System (OS/Docker/Kubernetes):
- Java Version (if you don't run kestra in Docker):

**Comment by prateekrai1** [comment]
Can I work on this issue?

**Comment by smunteankestra** [comment]
reproducing
<img width="1595" alt="image" src="https://github.com/user-attachments/assets/3da1306f-4ca7-4946-94d6-14cdb9d53274">
